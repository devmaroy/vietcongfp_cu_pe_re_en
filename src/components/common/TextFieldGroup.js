import React from 'react';
import PropTypes from 'prop-types';

const TextFieldGroup = ( { name, placeholder, value, error, info, type, onChange, disabled, className } ) => {
    return (
        <div>
            <input
                type={ type }
                placeholder={ placeholder }
                name={ name }
                value={ value }
                onChange={ onChange }
                disabled={ disabled }
                className={ className }
            />
            { info && <small>{ info }</small> }
            { error && <div>{ error }</div> }
        </div>
    );
};

TextFieldGroup.propTypes = {
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    value: PropTypes.string.isRequired,
    info: PropTypes.string,
    error: PropTypes.string,
    type: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    disabled: PropTypes.bool,
    className: PropTypes.string
};
  
TextFieldGroup.defaultProps = {
    type: 'text'
};

export default TextFieldGroup;