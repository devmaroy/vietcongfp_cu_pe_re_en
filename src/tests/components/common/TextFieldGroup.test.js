import React from 'react';
import { shallow } from 'enzyme';
import TextFieldGroup from '../../../components/common/TextFieldGroup';

test( 'should correctly render TextFieldGroup only with required props', () => {
    const wrapper = shallow(
            <TextFieldGroup 
                name="email"
                type="email"
                value="email@email.com"
                onChange={ () => {} }
                disabled={ false }
            />
        );
    
    expect( wrapper ).toMatchSnapshot();
});

test( 'should correctly render TextFieldGroup with all props', () => {
    const wrapper = shallow( 
            <TextFieldGroup 
                name="email"
                type="email"
                placeholder="Enter your email"
                value="emailemail.com"
                error="Email is not valid"
                info="This site uses Gravatar so if you want a profile image, use a Gravatar email"
                onChange={ () => {} }
                disabled={ false }
            /> 
        );

    expect( wrapper ).toMatchSnapshot();
});