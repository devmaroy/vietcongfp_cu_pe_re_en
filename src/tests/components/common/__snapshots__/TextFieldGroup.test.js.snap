// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render TextFieldGroup only with required props 1`] = `
<div>
  <input
    disabled={false}
    name="email"
    onChange={[Function]}
    type="email"
    value="email@email.com"
  />
</div>
`;

exports[`should correctly render TextFieldGroup with all props 1`] = `
<div>
  <input
    disabled={false}
    name="email"
    onChange={[Function]}
    placeholder="Enter your email"
    type="email"
    value="emailemail.com"
  />
  <small>
    This site uses Gravatar so if you want a profile image, use a Gravatar email
  </small>
  <div>
    Email is not valid
  </div>
</div>
`;
